process-create-data-repo-request

```diff
@@ -0,0 +1,253 @@
+# Start of execution
+# --------------------------------------------------------------------------------
+
+
+def main():
+    global opts, PENDING_APPROVAL
+    parser = OptionParser(usage="%prog <issue-id>")
+    parser.add_option(
+        "-n",
+        "--dry-run",
+        dest="dryRun",
+        action="store_true",
+        help="Do not post on Github",
+        default=False,
+    )
+    opts, args = parser.parse_args()
+
+    if len(args) != 1:
+        parser.print_help()
+        parser.error("Too many arguments")
+
+    GH_TOKEN = open(expanduser("~/.github-token")).read().strip()
+
+    issue_id = int(args[0])
+    gh = Github(login_or_token=GH_TOKEN)
+    api_rate_limits(gh)
+    cmssw_repo_name = "iarspider" + "/" + GH_CMSSW_REPO
+    cmssw_repo = gh.get_repo(cmssw_repo_name)
+    issue = cmssw_repo.get_issue(issue_id)
+    print("API Rate Limit")
+    print("Limit, Remaining: ", gh.rate_limiting)
+    print("Reset time (GMT): ", datetime.fromtimestamp(gh.rate_limiting_resettime))
+
+    # 0. Is this issue closed?
+    if issue.state == "closed":
+        print("Issue closed, ignoring.")
+        exit(0)
+
+    # 1. Is this a pull request?
+    if issue.pull_request:
+        print("This is a pull request, ignoring.")
+        exit(0)
+
+    title_match = re.match(CREATE_REPO, issue.title)
+
+    # 2. Is this issue meant to create a new cms-data repo?
+    if not title_match:
+        print("This issue is not for creating a repo, ignoring.")
+        print(issue.title)
+        exit(0)
+
+    category_name = title_match.group(1)
+    package_name = title_match.group(2)
+
+    print(category_name + "/" + package_name)
+
+    # 3. Does the requested repository already exist?
+    repo = None
+    try:
+        repo = gh.get_organization("cms-data").get_repo(
+            category_name + "-" + package_name
+        )
+    except github.UnknownObjectException:
+        pass
+
+    comments = [c for c in issue.get_comments()]
+
+    if repo:
+        post_message(issue, EXISTS_MSG.format(repo=repo.url), comments)
+        close_issue(issue)
+        exit(0)
+
+    # Figure out who must approve the action
+    data_categs = []
+    for cat, pkgs in CMSSW_CATEGORIES.items():
+        for pkg in pkgs:
+            if not pkg:
+                continue
+            if re.match(pkg + ".*", category_name + "/" + package_name):
+                data_categs.append(cat)
+                break
+
+    print(data_categs)
+
+    if not data_categs:
+        post_message(
+            issue,
+            INVALID_REQUEST_MSG.format(package=category_name + "/" + package_name),
+            comments,
+        )
+        exit(0)
+
+    # Get the status of this issue.
+    status = get_issue_status(issue)
+    print("Status: %s \n" % status)
+    print("Issue labels:", labels)
+
+    if status == NEW_ISSUE:
+        post_message(issue, "assign " + ",".join(data_categs), comments)
```

Comment:
please sort `data_categs` so that we get unique comment.

Reply:
done